pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFormatSpec.java

```diff
@@ -218,8 +261,9 @@ public String getSDFPattern() {
       dateTimeColumnValue = unit.convert(dateTimeColumnValueMS, TimeUnit.MILLISECONDS) / size;
     } else {
       String pattern = getSDFPattern();
+      DateTimeZone dateTimeZone = getDateTimezone();
```

Comment:
we should not invoke this method for every row as it will impact perf. time zone should be extracted once in the constructor and re-used.